### ISSUE TITLE
How to work with k8s CRDs

### ISSUE THREAD

**Issue by leoluk** [stripe/skycfg#70]
What's the right approach dealing with k8s CRDs in skycfg? Those are becoming increasingly common. cuelang solves this by having an intermediary typing language that can consume both Go structs and protobufs, such that it can validate either.

As far as I can tell, CRDs are only specified as jsonapi schemas and no protobuf schemas are generated.

**Comment by dilyevsky** [comment]
I haven’t tested it yet but If CRD provides openapi definitions you can
generate proto representations using google gnostic tool -
https://github.com/googleapis/gnostic-go-generator/blob/master/README.md

After that it should work the same way.

**Comment by leoluk** [comment]
How would one use `struct` objects for untyped objects?

On https://skycfg.fun, this does not work:

```
def main(ctx):
  return [struct()]
```

    `main' returned something that's not a protobuf (a struct)

**Comment by dilyevsky** [comment]
You can’t. Main needs to return proto wrapped into skyProtoMessage (there’s
a helper for that).

**Comment by jmillikin-stripe** [comment]
Do you know what Protobuf type Kubernetes is expecting to receive for a CRD? If it's something like [`google.protobuf.Struct`](https://github.com/protocolbuffers/protobuf/blob/v3.11.2/src/google/protobuf/struct.proto) then you can construct those directly in Skycfg using normal idioms.

**Comment by leoluk** [comment]
It appears that the k8s protobuf API does not support CRDs at all

**Comment by dilyevsky** [comment]
Correct, you have render them as json. Maybe that’ll change when go proto
v2 is out
